sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileAsyncApiTests.java

```diff
@@ -919,40 +913,39 @@ public void startCopyLeaseFail() {
     @MethodSource("com.azure.storage.file.share.FileShareTestHelper#startCopyArgumentsSupplier")
     public void startCopyWithOptions(boolean setFilePermissionKey, boolean setFilePermission, boolean ignoreReadOnly,
         boolean setArchiveAttribute, PermissionCopyModeType permissionType) {
-        primaryFileAsyncClient.create(1024).block();
         String sourceURL = primaryFileAsyncClient.getFileUrl();
-        String filePermissionKey = shareAsyncClient.createPermission(FILE_PERMISSION).block();
-        // We recreate file properties for each test since we need to store the times for the test with
-        // testResourceNamer.now()
-        smbProperties.setFileCreationTime(testResourceNamer.now()).setFileLastWriteTime(testResourceNamer.now());
-        if (setFilePermissionKey) {
-            smbProperties.setFilePermissionKey(filePermissionKey);
-        }
-        ShareFileCopyOptions options = new ShareFileCopyOptions().setSmbProperties(smbProperties)
-            .setFilePermission(setFilePermission ? FILE_PERMISSION : null)
-            .setIgnoreReadOnly(ignoreReadOnly)
-            .setArchiveAttribute(setArchiveAttribute)
-            .setPermissionCopyModeType(permissionType);
 
-        PollerFlux<ShareFileCopyInfo, Void> poller
-            = setPlaybackPollerFluxPollInterval(primaryFileAsyncClient.beginCopy(sourceURL, options, null));
+        StepVerifier.create(primaryFileAsyncClient.create(1024)
+            .then(shareAsyncClient.createPermission(FILE_PERMISSION))
+            .flatMapMany(filePermissionKey -> {
+                smbProperties.setFileCreationTime(testResourceNamer.now())
+                    .setFileLastWriteTime(testResourceNamer.now());
+                if (setFilePermissionKey) {
+                    smbProperties.setFilePermissionKey(filePermissionKey);
+                }
+                ShareFileCopyOptions options = new ShareFileCopyOptions().setSmbProperties(smbProperties)
+                    .setFilePermission(setFilePermission ? FILE_PERMISSION : null)
+                    .setIgnoreReadOnly(ignoreReadOnly)
+                    .setArchiveAttribute(setArchiveAttribute)
+                    .setPermissionCopyModeType(permissionType);
 
-        StepVerifier.create(poller)
+                return primaryFileAsyncClient.beginCopy(sourceURL, options, null);
```

Comment:
We still have to use `setPlaybackPollerFluxPollInterval` when calling beginCopy as it helps configure the options for playback mode, which is required for CI to function.